### PR TITLE
CLN/PERF: no need for kahan for int group_cumsum

### PR DESCRIPTION
**Pull request by mzeitlin11**
Surprised at lack of impact here - doesn't noticeably affect benchmarks.

Targeting the cython algo specifically shows an improvement (but smaller than I'd expect given the removed operations):
```
import numpy as np
import pandas._libs.groupby as libgroupby

N = 4_000_000
vals = np.random.randint(0, 10, (N, 5), dtype=np.int64)
result = np.empty_like(vals)

%timeit libgroupby.group_cumsum(result, vals, np.ones(N, dtype="int"), 1, False)
# 28.9 ms ± 805 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)  # this pr
# 37.3 ms ± 206 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)   # master
```

### CODE REVIEW

File: pandas/_libs/groupby.pyx [hunk 1]
```diff
@@ -253,18 +253,16 @@ def group_cumsum(numeric[:, ::1] out,
                         t = accum[lab, j] + y
                         compensation[lab, j] = t - accum[lab, j] - y
                         accum[lab, j] = t
-                        out[i, j] = accum[lab, j]
+                        out[i, j] = t
```
Comment (mzeitlin11) [inline pandas/_libs/groupby.pyx:256]: Doubt this affects compiled result, but may as well to not depend on a smart compiler

File: pandas/_libs/groupby.pyx [hunk 2]
```diff
@@ -253,18 +253,16 @@ def group_cumsum(numeric[:, ::1] out,
...
                     else:
                         out[i, j] = NaN
                         if not skipna:
                             accum[lab, j] = NaN
                             break
                 else:
-                    y = val - compensation[lab, j]
-                    t = accum[lab, j] + y
-                    compensation[lab, j] = t - accum[lab, j] - y
+                    t = val + accum[lab, j]
```
Comment (jreback) [inline pandas/_libs/groupby.pyx:263]: umm this is affecting all dtypes. do we not have tests for this for small floats?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/groupby.pyx (hunk 2 above)
Comment (mzeitlin11) [reply pandas/_libs/groupby.pyx:263]: This is inside the else block from an if statement `if numeric == float32_t or numeric == float64_t:` so only non-floats should end up here.

Comment (mzeitlin11) [reply pandas/_libs/groupby.pyx:265]: Have added a comment

**Comment by jreback** [comment]
hmm seemingly unrelated failures. maybe on master? cc @jbrockmendel

**Comment by mzeitlin11** [comment]
Looks like #41824

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mzeitlin11

**Pull request merged**